vendor/table_of_contents/lib/table_of_contents/heading.rb

```diff
@@ -0,0 +1,27 @@
+module TableOfContents
+  class Heading
+    def initialize(element_name:, text:, attributes:)
+      @element_name = element_name
+      @text = text
+      @attributes = attributes
+    end
+
+    def size
+      @element_name.scan(/h(\d)/) && $1 && Integer($1)
```

Comment:
This is more of a personal opinion than anything else, but I prefer avoiding `$1` et al and doing stuff to the `MatchData` class that's returned from `String.match`; something akin to

```ruby
@element_name.match(/h(\d)/)&.captures&.first&.to_i
``` 

Actually I don't know if that's any better. Meh.